### PR TITLE
Modified challenge phase query for displaying challenge phase detail

### LINKED ISSUE #1740: Modify challenge phase query for displaying challenge phase detail
**Current Scenario:**
The function [here](https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/views.py#L398) will **not** throw the error even if there are no challenge phase corresponding to the challenge.

**Steps to Reproduce:** You will only be able to check it only at the API level.
Let us suppose there exists a challenge with id 34 and phase ids 120, 121. Now when you will hit the [url ](https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/urls.py#L17)with challenge id 34 and phase id say 1 then it won't give error if phase id 1 exists(for any other challenge). 

Ideally, it this [url](https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/urls.py#L17) should only accept the phase ids 120 and 121 and should throw error for every other phase id.

**Deliverables:**
Modify the challenge phase query [here](https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/views.py#L406) to include the `challenge` parameter in it.

What should be added:
`challenge_phase = ChallengePhase.objects.get(challenge=challenge, pk=pk)`
Also, Update the response message to
`response_data = {'error': 'Challenge Phase {} does not exist'.format(pk)}`

### PR DESCRIPTION
**Pull request by HargovindArora**
This PR tends to fix #1740. @RishabhJain2018 Please review it and tell me if it needs some changes.

### CODE REVIEW

**Comment by HargovindArora** [comment]
@RishabhJain2018 Screenshots
I created a challenge phase belonging to challenge with `challenge_pk` = 1
Before making changes in views.py file I was able to access this challenge phase with `challenge_pk` =2 too:

screenshot from 2018-07-06 22-45-34

After making the changes in views.py file:

screenshot from 2018-07-06 22-45-41

File: tests/unit/challenges/test_views.py [hunk 1]
```diff
@@ -1398,6 +1398,19 @@ def test_get_challenge_phase_when_a_phase_is_not_public(self):
...
+                                        'pk': self.challenge_phase.pk + 10})
+
+        expected = {
+            'error': 'Challenge Phase {} does not exist'.format(self.challenge_phase.pk + 10)
+        }
+
+        response = self.client.get(self.url, {})
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_406_NOT_ACCEPTABLE)
+
```
Comment (RishabhJain2018) [inline tests/unit/challenges/test_views.py]: Please remove the above test case! It has been already added in the codebase.

**Comment by HargovindArora** [comment]
@deshraj @RishabhJain2018 Does this PR need any changes?

**Comment by RishabhJain2018** [comment]
No @HargovindArora This looks good to me! We will merge it shortly. Thanks a lot for the fix :)

**Comment by HargovindArora** [comment]
@RishabhJain2018 Okay :)

**Comment by deshraj** [comment]
@HargovindArora You have to rebase your branch and push again. See this documentation for git rebase: http://blog.algolia.com/master-git-rebase. Please feel free to ask if you have some doubts.

**Comment by HargovindArora** [comment]
@deshraj @RishabhJain2018  My system got crashed and I lost all the code part and thus my local branches and I had to reinstall my OS! I don't think I'll be able to do it now! 😞

**Comment by HargovindArora** [comment]
@RishabhJain2018 Okay got it. I'm on it. I'll ask if got into any trouble regarding rebasing! :sweat_smile:

**Comment by RishabhJain2018** [comment]
No, you didn't do it the right way. You have to rebase your branch with current master and force push it onto this branch.

**Comment by HargovindArora** [comment]
@RishabhJain2018 Okay! I'll do it again!

**Comment by RishabhJain2018** [comment]
Closing it due to inactivity!

**Pull request closed without merging**